demo/src/App.js

```diff
@@ -148,38 +147,51 @@ function App() {
     );
   });
 
-  const initialViewState =
-    sources[sourceName].initialViewState || DEFAULT_VIEW_STATE;
+  const viewState = {
+    ...(sources[sourceName].initialViewState || DEFAULT_VIEW_STATE),
+    height: viewHeight,
+    width: viewWidth,
+    id: 'detail'
+  };
+  const props = {
+    loader,
+    sliderValues: sliderValues.slice(
+      0,
+      sources[sourceName].channelNames.length
+    ),
+    colorValues: colorValues.slice(0, sources[sourceName].channelNames.length),
+    channelIsOn: channelIsOn.slice(
+      0,
+      sources[initSourceName].channelNames.length
+    ),
+    colormap: colormapOn
+  };
+  const detail = new DetailView({ viewState });
+  const views = [detail];
+  const layerProps = [props];
+  const viewStates = [viewState];
```

Comment:
This feels overly complicated to me and raises some serious concerns about how `viewState` is managed. Both `views` and `viewStates` are given to the `VivViewer` and both are maintaining the `viewState` somehow.